redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/ObservableMutableList.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.cash.redwood
+
+internal class ObservableMutableList<T>(
+  private val delegate: MutableList<T> = mutableListOf(),
+  private val onChange: () -> Unit = {},
+  // Delegate to List-only so new mutable methods have to be implemented manually.
+) : MutableList<T>, List<T> by delegate {
```

Comment:
Does `AbstractMutableList` help reduce needing to implement some of these functions yourself?

Reply:
Yep! It works great, thanks.